proposals/buffer-view.md

```diff
@@ -0,0 +1,190 @@
+# Buffer View
+
+* Status: **Draft**
+* Last modified: 2025-10-20
+* Issue: [#5338](https://github.com/gpuweb/gpuweb/issues/5338)
+
+# Overview
+
+This proposal intends to add a new opaque host-shareable type to WGSL and two
+new built-in functions to interact with the type.
+The purpose of the type is to provide a limited form of type-punning support in
+WGSL.
+That is, to support reinterpreting data in a buffer as various different types.
+
+This can also serve as a solution to a similar sort of request for workgroup
+memory: [#5136](https://github.com/gpuweb/gpuweb/issues/5156).
+
+New functionality will be added under the language extension: `buffer_view`.
+
+# Buffer Type
+
+Add a new opaque host-shareable type to WGSL: `buffer<N>`.
+This type can be the effective-value-type of a variable in the storage,
```

Comment:
Isn't it sufficient to just say, "can be the type of a variable in the storage"? How is the distinction important in this context?

Reply:
In the var declarations section of the spec the type restrictions are described in terms of the effective-value-type so I just reused that here. You're right it is not strictly necessary since this is not an abstract type.